python/lsst/afw/geom/testUtils.py

```diff
@@ -90,3 +99,757 @@ def __iter__(self):
                 yield self[col, row]
 
 
+class TransformTestBaseClass(lsst.utils.tests.TestCase):
+    """Base class for unit tests of Transform<X>To<Y> and subclasses
+
+    Subclasses must call `TransformTestBaseClass.setUp(self)`
+    if they provide their own version.
+
+    If a package other than afw uses this class then it must
+    override the `getTestDir` method to avoid writing into
+    afw's test directory.
+    """
+
+    def getTestDir(self):
+        """Return a directory where temporary test files can be written
+
+        This is typically the test dir of the package in which the test lives
+        """
+        return os.path.join(lsst.utils.getPackageDir("afw"), "tests")
+
+    def setUp(self):
+        # tell unittest to use the msg argument of asserts as a supplement
+        # to the error message, rather than as the whole error message
+        self.longMessage = True
+
+        # list of endpoint class name prefixes; the full name is prefix + "Endpoint"
+        self.endpointPrefixes = ("Generic", "Point2", "Point3", "SpherePoint")
+
+        # GoodNaxes is dict of endpoint class name prefix:
+        #    tuple containing 0 or more valid numbers of axes
+        self.goodNaxes = {
+            "Generic": (1, 2, 3, 4),  # all numbers of axes are valid for GenericEndpoint
+            "Point2": (2,),
+            "Point3": (3,),
+            "SpherePoint": (2,),
+        }
+
+        # BadAxes is dict of endpoint class name prefix:
+        #    tuple containing 0 or more invalid numbers of axes
+        self.badNaxes = {
+            "Generic": (),  # all numbers of axes are valid for GenericEndpoint
+            "Point2": (1, 3, 4),
+            "Point3": (1, 2, 4),
+            "SpherePoint": (1, 3, 4),
+        }
+
+    @staticmethod
+    def makeRawArrayData(nPoints, nAxes, delta=0.123):
+        return np.array([[i + j*delta for j in range(nAxes)] for i in range(nPoints)])
+
+    @staticmethod
+    def makeRawPointData(nAxes, delta=0.123):
+        return [i*delta for i in range(nAxes)]
+
+    @staticmethod
+    def makeEndpoint(name, nAxes):
+        """Make an endpoint
+
+        Parameters
+        ----------
+        name : string
+            one of "Generic", "Point2", "Point3" or "SpherePoint"
+        nAxes : integer
+            number of axes; ignored if the name is not "Generic"
```

Comment:
I'll use None. I also found that 0 was acceptable and added code to Endpoint to prohibit non-positive values.